pandas/tests/arrays/categorical/test_replace.py

```diff
@@ -68,7 +70,11 @@ def test_replace_categorical(to_replace, value, result, expected_error_msg):
             # ensure non-inplace call does not affect original
             tm.assert_categorical_equal(cat, expected)
 
-    pd.Series(cat, copy=False).replace(to_replace, value, inplace=True)
+    if using_copy_on_write:
+        with tm.raises_chained_assignment_error():
+            pd.Series(cat, copy=False).replace(to_replace, value, inplace=True)
+    else:
+        pd.Series(cat, copy=False).replace(to_replace, value, inplace=True)
     tm.assert_categorical_equal(cat, expected)
```

Comment:
In this case, I assume we are not necessarily testing that it is done on a single line, but just that the inplace works (and updates the underlying array. So I think it might be better to change the test to:

```
ser = pd.Series(cat, copy=False)
ser.replace(to_replace, value, inplace=True)
```

which should then work the same for both CoW and not?

Reply:
Don't care much either way, updated the test